src/components/task-item/bar/custom-milestone.tsx

```diff
@@ -0,0 +1,77 @@
+import React from "react";
+import { getProgressPoint } from "../../../helpers/bar-helper";
+import { BarDateHandle } from "./bar-date-handle";
+import { BarProgressHandle } from "./bar-progress-handle";
+import { TaskItemProps } from "../task-item";
+import styles from "./bar.module.css";
+import { CustomMilestoneDisplay } from "../custom-milestone/custom-milestone-display";
+
+export const CustomMilestone: React.FC<TaskItemProps> = ({
+  task,
+  isProgressChangeable,
+  isDateChangeable,
+  rtl,
+  onEventStart,
+  isSelected,
+}) => {
+  const progressPoint = getProgressPoint(
+    +!rtl * task.progressWidth + task.progressX,
```

Comment:
Maybe easier to read and understand
```suggestion
    (rtl ? 0 : 1) * task.progressWidth + task.progressX,
```